boring-sys/build.rs

```diff
@@ -288,6 +288,14 @@ fn get_extra_clang_args_for_bindgen() -> Vec<String> {
             params.push("-isysroot".to_string());
             params.push(sysroot);
         }
+        "android" => {
+            let android_ndk_home = std::env::var("ANDROID_NDK_HOME")
+                .expect("Please set ANDROID_NDK_HOME for Android build");
+            let mut android_sysroot = std::path::PathBuf::from(android_ndk_home);
+            android_sysroot.push("sysroot");
+            params.push("--sysroot".to_string());
+            params.push(android_sysroot.to_string_lossy().into());
```

Comment:
to_string_lossy looks wrong. This should either be an `unwrap` (I doubt anyone is using non-unicode paths in practice) or we should pass byte strings to the bindgen invocation.

Reply:
Bindgen doesn't accept byte strings. :sob: I'll make it an `unwrap`, or rather an `expect`; you're right that substituting U+FFFD is never going to make a valid path and only obscures the issue.

Reply:
Even better, `std::env::var` already checked for us.